pkg/object/websocketserver/spec.go

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package websocketserver
+
+import (
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"strings"
+)
+
+type (
+	// Spec describes the WebSocketServer.
+	Spec struct {
+		Port    uint16 `yaml:"port" jsonschema:"required,minimum=1"`
+		HTTPS   bool   `yaml:"https" jsonschema:"required"`
+		Backend string `yaml:"backend" jsonschema:"required"`
+
+		CertBase64 string `yaml:"certBase64" jsonschema:"omitempty,format=base64"`
+		KeyBase64  string `yaml:"keyBase64" jsonschema:"omitempty,format=base64"`
+
+		wssCertBase64 string `yaml:"certBase64" jsonschema:"omitempty,format=base64"`
+		wssKeyBase64  string `yaml:"keyBase64" jsonschema:"omitempty,format=base64"`
```

Comment:
is it need to bind multiple certs? as #31

Reply:
As the implementation right now:
1. Before protocol switching, there is only one HTTPServer for handling the traffic. 
2. After protocol switching, the `WebScoket` server only accepts one routable backend. 

So I think maybe it's not necessary to introduce multiple certs support in this version.